tools/configuration-converter.sh

```diff
@@ -0,0 +1,126 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script is used to convert istio configuration to sail operator configuration.
+# In the end of the execution new yaml file will be created with "sail-ISTIO_CONFIG_YAML" name.
+# Usage: ./configuration-converter.sh ISTIO_CONFIG_YAML_WITH_PATH, example:  ./configuration-converter.sh sample_config.yaml"
+set -e
+
+# Default values
+NAMESPACE="istio-system"
+VERSION=""
+INPUT=""
+OUTPUT=""
+
+# Function to show usage
+usage() {
+    echo "Usage: $0 <input> [output] [-n <namespace>] [-v <version>]"
+    echo "  <input>         : Input file (required)"
+    echo "  [output]        : Output file (optional, defaults to input's base directory with '-sail.yaml' suffix)"
+    echo "  -n <namespace>  : Namespace (optional, defaults to 'istio-system')"
+    echo "  -v <version>    : Istio Version (required)"
+    exit 1
+}
+
+# Ensure input file is provided
+if [[ -z "$1" ]]; then
+    echo "Error: Input file is required."
+    usage
+fi
+
+INPUT="$1"
+OUTPUT="$2"
+
+# Shift parameters if output is provided
+if [[ -n "$OUTPUT" && "$OUTPUT" != -* ]]; then
+    shift 2
+else
+    OUTPUT="$(dirname "$INPUT")/$(basename "$INPUT" .yaml)-sail.yaml"
+    shift 1
+fi
+
+# Ensure output is not a directory
+if [[ -d "$OUTPUT" ]]; then 
+    echo "Error: OUTPUT must be a file, not a directory."
+    exit 1
+fi
+
+# Parse optional namespace and version arguments
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -n)
+            NAMESPACE="$2"
+            shift 2
+            ;;
+        -v)
+            VERSION="$2"
+            shift 2
+            ;;
+        *)
+            usage
+            ;;
+    esac
+done
+
+if [[ -z "$VERSION" ]]; then
+    echo "Error: Version (-v) is required."
+    usage
+fi
+
+if ! command -v yq &>/dev/null; then
+    echo "Error: 'yq' is not installed. Please install it before running the script."
+    exit 1
+fi
+
+function add_mandatory_fields(){
+    yq -i eval ".apiVersion = \"sailoperator.io/v1\" 
+    | .kind = \"Istio\" 
+    | (select(.spec.meshConfig) | .spec.values.meshConfig) = .spec.meshConfig 
+    | (select(.spec.values.istio_cni) | .spec.values.pilot.cni) = .spec.values.istio_cni 
+    | .metadata.name = \"default\" 
+    | .spec.version = \"$VERSION\" 
+    | .spec.namespace = \"$NAMESPACE\" 
+    | del(.spec.values.istio_cni)
+    | del(.spec.meshConfig)
+    | del(.spec.hub) 
+    | del(.spec.tag) 
+    | del(.spec.values.gateways)" "$OUTPUT"
+}
+
+#Convert boolean values to string if they are under *.env
+function boolean_2_string(){
+    yq -i -e '(.spec.values.[].env.[] | select(. == true)) |= "true" 
+    | (.spec.values.[].env.[] | select(. == false)) |= "false"' "$OUTPUT" 
+    yq -i -e 'del(.. | select(tag == "!!seq" and length == 0))' "$OUTPUT"
```

Comment:
I like how this script is so much simpler now. Can we merge these two commands into one?